youtube_dl/extractor/flipagram.py

```diff
@@ -0,0 +1,114 @@
+        user_data = self._parse_json(re.search(r'window.reactH2O\s*=\s*({.+});', webpage).group(1), video_id)
+
+        thumbnails = []
+        i = 0
+        for cover in user_data.get('flipagram').get('covers'):
+            thumbnails.append({
+                'url': 'https:' + cover.get('url'),
+                'width': cover.get('width'),
+                'height': cover.get('height'),
+            })
+            i += 1
+
+        comments = []
+        i = 0
+        for comment in user_data.get('comments').get(video_id).get('items'):
+            comments.append({
+                'author': comment.get('user').get('name'),
+                'author_id': comment.get('user').get('username'),
+                'id': comment.get('id'),
+                'text': comment.get('comment')[0],
+                #'timestamp': comment[i].get('created'), # convert '04/25/2016 00:23:24 UTC' to UNIX timestamp
+                'parent': 'root'
+            })
+
+        tags = []
+        i = 1
+        for tag in user_data.get('flipagram').get('story'):
+            if isinstance(type(tag), dict):
+                try:
+                    tags.append(tag.itervalues().next())
+                except AttributeError:
+                    # Python 3.x
+                    tags.append(next(iter(tag.values())))
+
+        formats = []
+
+        try:
+            audio_url = user_data.get('flipagram').get('music').get('track').get('previewUrl')
+            formats.append({
+                'url': audio_url,
+                'ext': 'm4a',
+                'format_note': '(audio only)',
+            })
+        except AttributeError:
+            pass
+
+        formats.append({
+            'url': content_data.get('contentUrl'),
+            'ext': 'mp4',
+            'width': user_data.get('flipagram').get('video').get('width'),
+            'height': user_data.get('flipagram').get('video').get('height'),
+        })
+
+        return {
+            'id': video_id,
+            'title': content_data.get('name'),
+            'formats': formats,
+            'ext': 'mp4',
+            'thumbnails': thumbnails,
+            'description': content_data.get('description'),
+            'uploader': user_data.get('flipagram').get('user').get('name'),
+            'creator': user_data.get('flipagram').get('user').get('name'),
+            #'timestamp': parse_iso8301(user_data.get('flipagram').get('iso8601Created')),
+            #'upload_date': unified_strdate(user_data.get('flipagram').get('created')),
+            'uploader_id': user_data.get('flipagram').get('user').get('username'),
+            'uploader_url': 'https://flipagram.com' + user_data.get('flipagram').get('user').get('username'),
+            'view_count': user_data.get('flipagram').get('counts').get('plays'),
+            'repost_count': user_data.get('flipagram').get('counts').get('reflips'),
+            'comment_count': user_data.get('flipagram').get('counts').get('comments'),
```

Comment:
Use temporary variables to avoid duplicated ```user_data.get('flipagram').get('counts')``` and ```user_data.get('flipagram').get('user')``` patterns. And use ```[]``` as the fallback in ```.get()```.